design-system/react/src/components/Dialog/defs.ts

```diff
@@ -0,0 +1,183 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import type { ReactNode } from 'react';
+import { createContext, useContext } from 'react';
+import type { AriaOverlayProps, AriaDialogProps, ModalAria } from 'react-aria';
+import type { OverlayTriggerState } from 'react-stately';
+
+import type { ButtonProps } from '../Button';
+
+import type { DialogClose } from './DialogClose';
+import type { DialogContent } from './DialogContent';
+import type { DialogDescription } from './DialogDescription';
+import type { DialogFooter } from './DialogFooter';
+import type { DialogHeading } from './DialogHeading';
+import type { DialogTrigger } from './DialogTrigger';
+
+import type { Components } from '~/defs';
+import type { CreateComponent, HTMLProps } from '~/utils';
+
+// ----------------------------------------------------------------------------
+// Context
+// ----------------------------------------------------------------------------
+
+export type DialogContext = {
+  state: OverlayTriggerState;
+  triggerRef?: React.MutableRefObject<HTMLDivElement | null>;
+  overlayProps?: React.HTMLAttributes<HTMLElement>;
+  modalProps?: ModalAria['modalProps'];
+  dialogProps?: React.HTMLAttributes<HTMLElement>;
+  headingProps?: React.HTMLAttributes<HTMLElement>;
+  isBlocked?: boolean;
+};
+
+export const DialogCtx = createContext<DialogContext>({} as DialogContext);
```

Comment:
since this file should have just type, you can move this `DialogCtx` and the `useDialog` to the `Dialog.tsx` itself instead